### PR TITLE
feat: add telegram bot

### PR DESCRIPTION
**Pull request by vBrusentsov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return ctx.reply("Please enter a number")
+    }
+    ctx.state.arguments = ctx.match.split(' ');
+    const validateNumbers = ctx.state.arguments.some((elem) => {
+        return Number.isNaN(Number(elem))
+    })
+
+    if(validateNumbers === true){
+        return ctx.reply("Incorrect number entered")
+    } else {
```
Comment (jaxin007) [inline index.js]: You could remove else block here

File: index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return next()
+    }
+}
+
+bot.command("dimagay", (ctx)=> ctx.reply("WHY ARE YOU GAy"));
+bot.command("start", (ctx) => ctx.reply("welcome! "));
+bot.command("stop", (ctx) => ctx.reply("Okay. I`m will sleep"));
+bot.command("add",middlewareMatchToSplit, (ctx) => {
+    let result = ctx.state.arguments.reduce((previousValue, currentValue) => {
```
Comment (jaxin007) [inline index.js]: Why do you use let instead of const? I dont see any place where this value may be changed

File: index.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        return +previousValue + +currentValue
+    })
+    return ctx.reply(result);
+})
+
+bot.command("subtract",middlewareMatchToSplit, (ctx) => {
+    let result = ctx.state.arguments.reduce((previousValue, currentValue) => {
```
Comment (jaxin007) [inline index.js]: Same

File: index.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        return +previousValue - +currentValue
+    })
+    return ctx.reply(result);
+})
+
+
+bot.command("multiply",middlewareMatchToSplit,  (ctx) => {
+    let result = ctx.state.arguments.reduce((previousValue, currentValue) => {
```
Comment (jaxin007) [inline index.js]: Same

File: index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+})
+
+bot.hears(/^\/.*/, (ctx) => {
+    ctx.reply("Please use one of the following commands: \n/add\n/subtract\n/multiply\n/start\n/stop ")
+});
+
+
+bot.hears("Hello", async (ctx) => {
+    await ctx.reply("Hello", {
+        reply_to_message_id: ctx.msg.message_id,
```
Comment (jaxin007) [inline index.js]: Previously you haven't used async/await in your "controllers", so i'd recommend you to follow one style and just return your statement

File: package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "telegrambot",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "dependencies": {
+    "grammy": "^1.9.1"
+  },
+  "devDependencies": {},
+  "scripts": {
```
Comment (jaxin007) [inline package.json:10]: Where's start command?

**Review by jaxin007 — CHANGES_REQUESTED** [review]

File: index.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+const middlewareMatchToSplit = function (ctx, next){
+    if(ctx.match === ''){
+        return ctx.reply("Please enter a number")
+    }
+    ctx.state.arguments = ctx.match.split(' ');
+    const validateNumbers = ctx.state.arguments.some((elem) => {
+        return Number.isNaN(Number(elem))
+    })
+
+    if(validateNumbers === true){
```
Comment (jaxin007) [inline index.js]: Why strict checking for true? You could keep only if (validateNumbers) and it will work for you

File: index.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
+const { Bot } = require('grammy');
+
+const bot = new Bot('5532232117:AAG_8EgKVkKd04srN7B5aHLKatC3biacIHc');
+bot.use((ctx, next) => {
+    ctx.state = {};
+    return next();
+})
+
+const middlewareMatchToSplit = function (ctx, next){
+    if(ctx.match === ''){
```
Comment (jaxin007) [inline index.js]: You could check for falsy boolean here instead of empty string

File: index.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
+const { Bot } = require('grammy');
+
+const bot = new Bot('5532232117:AAG_8EgKVkKd04srN7B5aHLKatC3biacIHc');
```
Comment (jaxin007) [inline index.js]: I wouldn't recommend you to do this. Use env variables and remove all secrets from github that may compromise you

File: index.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    return next();
+})
+
+const middlewareMatchToSplit = function (ctx, next){
+    if(ctx.match === ''){
+        return ctx.reply("Please enter a number")
+    }
+    ctx.state.arguments = ctx.match.split(' ');
+    const validateNumbers = ctx.state.arguments.some((elem) => {
+        return Number.isNaN(Number(elem))
```
Comment (jaxin007) [inline index.js]: A few lines below you're using + statement to convert value to number. Then why use Number here?

**Review by jaxin007 — CHANGES_REQUESTED** [review]

File: index.js (hunk 9 above)
Comment (DrSmile444) [reply index.js]: In addition, you must revoke your token and generate a new one using BotFather with the .env file.

File: index.js (hunk 7 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 8 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 9 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 10 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+dotenv.config();
+
+const bot = new Bot(process.env.BOT_TOKEN);
+
+bot.use((ctx, next) => {
+    ctx.state = {};
+    return next();
+})
+
+const middlewareMatchToSplit = function (ctx, next){
```
Comment (DrSmile444) [inline index.js]: `function (ctx, next){`
No space between ) and {

File: index.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    if(!ctx.match){
```
Comment (DrSmile444) [inline index.js]: No space between ) {

File: index.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+        return ctx.reply("Please enter a number")
```
Comment (DrSmile444) [inline index.js]: No semicolon

File: index.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    }
```
Comment (DrSmile444) [inline index.js:17]: No additional space after the block

File: index.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+    ctx.state.arguments = ctx.match.split(' ');
+    const validateNumbers = ctx.state.arguments.some((elem) => {
+        return Number.isNaN(+(elem))
```
Comment (DrSmile444) [inline index.js]: Extra parentheses that wrap elem

File: index.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    })
+
+    if(validateNumbers){
```
Comment (DrSmile444) [inline index.js]: No spaces between if and (, ) and {

File: index.js [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+        return ctx.reply("Incorrect number entered")
```
Comment (DrSmile444) [inline index.js]: No semicolon

File: index.js [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+    }
```
Comment (DrSmile444) [inline index.js:27]: No additional space after the block ends

File: index.js [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+        return next()
```
Comment (DrSmile444) [inline index.js]: Extra tab, no semicolon

File: index.js [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+bot.command("dimagay", (ctx)=> ctx.reply("WHY ARE YOU GAy"));
+bot.command("start", (ctx) => ctx.reply("welcome! "));
+bot.command("stop", (ctx) => ctx.reply("Okay. I`m will sleep"));
+bot.command("add",middlewareMatchToSplit, (ctx) => {
```
Comment (DrSmile444) [inline index.js]: No space between , and middlewareMatchToSplit

File: index.js [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+    const result = ctx.state.arguments.reduce((previousValue, currentValue) => {
+        return +previousValue + +currentValue
```
Comment (DrSmile444) [inline index.js]: No semicolon. Review all your code to add semicolons

File: index.js [hunk 22]
```diff
@@ -0,0 +1,64 @@
+const dotenv = require('dotenv')
+const { Bot } = require('grammy');
+
+dotenv.config();
+
+const bot = new Bot(process.env.BOT_TOKEN);
```
Comment (DrSmile444) [inline index.js]: Wrap your logic into async IIFE

**Review by DrSmile444 — CHANGES_REQUESTED** [review]

File: index.js (hunk 11 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 12 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 11 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 13 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 14 above)
Comment (vBrusentsov) [reply index.js:17]: Done

File: index.js (hunk 15 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 16 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 17 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 18 above)
Comment (vBrusentsov) [reply index.js:27]: Done

File: index.js (hunk 19 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 20 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 21 above)
Comment (vBrusentsov) [reply index.js]: Done

File: index.js (hunk 22 above)
Comment (vBrusentsov) [reply index.js]: Done

**Review by DrSmile444 — APPROVED** [review]

**Review by DrSmile444 — APPROVED** [review]

**Pull request merged**